src/Files.App/Views/Settings/ActionsPage.xaml.cs

```diff
@@ -0,0 +1,206 @@
+				ViewModel.ShowAddNewShortcutGrid = false;
+
+				if (ViewModel.SelectedNewShortcutItem is not null)
+				{
+					ViewModel.SelectedNewShortcutItem.HotKeyText = "";
+					ViewModel.SelectedNewShortcutItem = null;
+				}
+
+
+				ViewModel.KeyboardShortcuts.ForEach(x => x.IsEditMode = false);
+				item.IsEditMode = true;
+			}
+		}
+
+		private void SaveButton_Click(object sender, RoutedEventArgs e)
+		{
+			if (sender is not Button button || button.DataContext is not RichCommandShortcutItem item)
+				return;
+
+			item.IsEditMode = false;
+
+			if (item.HotKeyText == item.PreviousHotKeyText)
+				return;
+
+			var actions = new Dictionary<string, string>(GeneralSettingsService.Actions);
+
+			// Remove existing setting
+			foreach (var action in actions)
+			{
+				if (Enum.TryParse(action.Key, true, out CommandCodes code) && code == item.CommandCode)
+					actions.Remove(action.Key);
+			}
+
+			if (item.HotKeyText == item.DefaultHotKeyText)
+				return;
+
+			// Create a new one
+			actions.Add(item.CommandCode.ToString(), item.HotKeyText);
+
+			// Set
+			item.HotKeys = new(HotKeyCollection.Parse(item.HotKeyText));
+			GeneralSettingsService.Actions = actions;
+		}
+
+		private void ResetButton_Click(object sender, RoutedEventArgs e)
+		{
+			if (sender is Button button && button.DataContext is RichCommandShortcutItem item)
+			{
+				item.IsEditMode = false;
+
+				var actions = new Dictionary<string, string>(GeneralSettingsService.Actions);
+
+				// Remove existing setting
+				foreach (var action in actions)
+				{
+					if (Enum.TryParse(action.Key, true, out CommandCodes code) && code == item.CommandCode)
+						actions.Remove(action.Key);
+				}
+
+				// Set
+				item.HotKeys = new(HotKeyCollection.Parse(item.DefaultHotKeyText));
+				GeneralSettingsService.Actions = actions;
+			}
+		}
+
+		private void EditorTextBox_PreviewKeyDown(object sender, KeyRoutedEventArgs e)
+		{
+			if (sender is not TextBox textBox)
+				return;
+
+			var pressedKey = e.OriginalKey;
+
+			// If pressed key is one of modifier don't show it in the TextBox yet
+			foreach (var modifier in ModifierKeys)
+			{
+				if (pressedKey == modifier)
+					return;
+			}
+
+			var modifiers = VirtualKeyModifiers.None;
+
+			// TODO: Move to HotKeyHelpers.GetCurrentKeyModifiers()
+			foreach (var modifier in ModifierKeys)
+			{
+				var keyState = InputKeyboardSource.GetKeyStateForCurrentThread(modifier);
+
+				if (keyState.HasFlag(CoreVirtualKeyStates.Down))
+				{
+					switch (modifier)
+					{
+						case VirtualKey.Control:
+						case VirtualKey.LeftControl:
+						case VirtualKey.RightControl:
+							modifiers |= VirtualKeyModifiers.Control;
+							break;
+						case VirtualKey.Menu:
+						case VirtualKey.LeftMenu:
+						case VirtualKey.RightMenu:
+							modifiers |= VirtualKeyModifiers.Menu;
+							break;
+						case VirtualKey.Shift:
+						case VirtualKey.LeftShift:
+							modifiers |= VirtualKeyModifiers.Shift;
+							break;
+					}
+				}
+			}
```

Comment:
I'd like to know if it was on purpose that HotKeyHelpers.GetCurrentKeyModifier has smaller switch expression than this such as not having LeftShift, RightShift.
@0x5ba Should we add them too like this?

Reply:
@cinqmilleans

Reply:
Also I'd like to know why "!" is replaced with " !" in HotKeyCollection line38